airflow/policies.py

```diff
@@ -117,6 +117,19 @@ def get_dagbag_import_timeout(dag_file_path: str) -> int | float:  # type: ignor
     """
 
 
+@local_settings_hookspec
+def connection_policy(conn_id: str) -> str:  # type: ignore[empty-body]
```

Comment:
I thought for a second that maybe we should call this `connection_id_policy` or something because... e.g. with task policy or dag policy, you get and can mutate the actual object -- but here we only touch the conn_id.  So the naming is perhaps slightly inconsistent but that's nit picking.

I suppose we could call it `connection_retrieval_policy`?  Or leave it 🤷